sygnal/apnspushkin.py

```diff
@@ -79,9 +79,18 @@ class ApnsPushkin(ConcurrencyLimitedPushkin):
     Relays notifications to the Apple Push Notification Service.
     """
 
-    # Errors for which the token should be rejected
-    TOKEN_ERROR_REASON = "Unregistered"
-    TOKEN_ERROR_CODE = 410
+    # Errors for which the token should be rejected and not reused
+    # See https://developer.apple.com/documentation/usernotifications/setting_up_a_remote_notification_server/handling_notification_responses_from_apns  # noqa: E501
+    # for the full list of possible errors.
+    TOKEN_ERRORS = {
+        (400, "BadDeviceToken"),  # Invalid token
+        # `DeviceTokenNotForTopic` may be due to a bad token or a Sygnal
+        # misconfiguration. It's alright for the homeserver to forget a valid token,
+        # since iOS clients are expected to re-upload their device tokens periodically.
+        (400, "DeviceTokenNotForTopic"),
+        (400, "TopicDisallowed"),
```

Comment:
Do both of these mean a broken client?

Reply:
I think both of these mean a developer has messed up in some way. I'm not 100% clear on the how. I think the former means the token is valid but for a different app ID and the latter seems to be something to do with mixing up VOIP and non-VOIP tokens/topics.

Reply:
Alright! Thanks for the explanation!